lib/sucker_punch/job.rb

```diff
@@ -1,25 +1,64 @@
 module SuckerPunch
+  # Include this module in your job class
+  # to create asynchronous jobs:
+  #
+  # class LogJob
+  #   include SuckerPunch::Job
+  #   workers 4
+  #
+  #   def perform(*args)
+  #     # log the things
+  #   end
+  # end
+  #
+  # To trigger asynchronous job:
+  #
+  #   LogJob.perform_async(1, 2, 3)
+  #
+  # Note that perform_async is a class method, perform is an instance method.
   module Job
     def self.included(base)
-      base.send(:include, ::Celluloid)
       base.extend(ClassMethods)
+      base.class_attribute :num_workers
 
-      base.class_eval do
-        def self.new
-          define_celluloid_pool(self, @workers)
-        end
-      end
+      base.num_workers = 2
+    end
+
+    def logger
+      SuckerPunch.logger
     end
 
     module ClassMethods
+      def perform_async(*args)
+        return unless SuckerPunch::RUNNING.true?
```

Comment:
Looking back at this code I now realize I've created a race condition. The gem's shutdown function may potentially be called while the `post_async` function is in-progress. There is no guarantee that post_async finishes before the shutdown handler. Therefore the thread pool could potentially be shutdown *before* the job actually gets post. That isn't really a problem given the current implementation (the post on line 35 would simply be rejected by the pool) but it's technically incorrect. I noticed this when I was looking at ways to track job completion on shutdown. The approach I was thinking of taking won't work with this race condition.

Now that you've described the shutdown behavior you want I'm revisiting this. The simplest option is to create a single lock and pass all post/shutdown action through it, but that seems heavy-handed. I think we can do better. I'm working on it.